### PR TITLE
[DDW-429] Add Network Prefix to Node IPC Local Storage Keys and Process Names

### PR DESCRIPTION
**Pull request by MarcusHurney**
- Adds dynamic prefix derivation to local storage keys in CardanoNode based on the current network.

- Adds dynamic derivation of process names for cardano-node based on the current platform.

- Defines flow types for network names, process names, and local storage key names.

### Basics

- [ ] PR is updated to the most recent version of target branch (and there are no conflicts)
- [ ] PR has good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance tests are passing (`yarn run test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn run dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn run package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn run flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn run lint`)
- [ ] Text changes are proofread and approved (Jane Wild)
- [ ] There are no missing translations (running `yarn run manage:translations` produces no changes)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn run storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly documented and commented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-rendering
- [ ] Any code that only works in Electron is neatly separated from components

### Testing
- [ ] New feature / change is covered by acceptance tests
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature / change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review:
- [ ] Merge PR
- [ ] Delete source branch
- [ ] Move ticket to `done` on the Youtrack board

### CODE REVIEW

File: source/common/types/cardanoNode.types.js [hunk 1]
```diff
@@ -24,9 +42,64 @@ export const CardanoNodeStates: {
...
+  'development-PREVIOUS-DAEDALUS-PID' |
+  string
+);
+
+export type CardanoNodeStorageKeys = {
+  PREVIOUS_CARDANO_PID: CardanoPidOptions,
+  PREVIOUS_DAEDALUS_PID: DaedalusPidOptions
+};
+
+// TODO: Add more options when the exact process names are established
```
Comment (nikolaglumac) [inline source/common/types/cardanoNode.types.js]: @cleverca22 we need your help with this part.
If I am not mistaking we have:
- `Electron` as process name in development AND in the Linux production, and
- `Daedalus` as process name in production for OSX and Win
Is that right?

File: source/common/types/cardanoNode.types.js [hunk 2]
```diff
@@ -24,9 +42,64 @@ export const CardanoNodeStates: {
...
+export type DaedalusProcessNames = (
+  'Electron'
+);
+
+// TODO: Determine if the cardano-node can be named anything else
```
Comment (nikolaglumac) [inline source/common/types/cardanoNode.types.js]: @cleverca22 same thing here - is the Cardano Node process always called `cardano-node` on all of the platforms?

**Comment by DominikGuzei** [comment]
As discussed with @nikolaglumac  i think it would be better to use a lock file instead of the fragile naming conventions for processes on the various operating systems.

**Comment by nikolaglumac** [comment]
@DominikGuzei but we should still keep tracking the previous Cardano PID right?
Only Daedalus PID can be removed?

**Comment by DominikGuzei** [comment]
Yeah, this only applies to Daedalus - the name of cardano-node process seems to be pretty standard across any builds?

**Comment by nikolaglumac** [comment]
@DominikGuzei we are waiting on a confirmation from @cleverca22 regarding that...

@MarcusHurney please remove all stuff related to the PREVIOUS_DAEDALUS_PID - including the name types and functions. Thanks!

File: source/common/types/cardanoNode.types.js (hunk 1 above)
Comment (nikolaglumac) [reply source/common/types/cardanoNode.types.js]: @cleverca22 don't waste time on this one - we will remove this part of the code...

File: source/common/types/cardanoNode.types.js (hunk 2 above)
Comment (nikolaglumac) [reply source/common/types/cardanoNode.types.js]: @cleverca22 this is still important to know!

**Comment by MarcusHurney** [comment]
@nikolaglumac 
Ok, everything relating to `PREVIOUS_DAEDALUS_PID` has been removed.

File: source/common/types/cardanoNode.types.js (hunk 2 above)
Comment (cleverca22) [reply source/common/types/cardanoNode.types.js]: on windows it will be `cardano-node.exe` and on linux and darwin, just `cardano-node`

Comment (nikolaglumac) [reply source/common/types/cardanoNode.types.js]: Thanks @cleverca22!
@MarcusHurney please update the code to match this 🙏

Comment (MarcusHurney) [reply source/common/types/cardanoNode.types.js]: @nikolaglumac 
I've added it!

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,7 @@ Changelog
 
 ### Chores
 
+- Adds dynamic prefix derivation to local storage keys in CardanoNode based on the current network. Adds dynamic derivation of process names for both Daedalus and cardano-node based on the current network. [PR 1109](https://github.com/input-output-hk/daedalus/pull/1109)
```
Comment (DominikGuzei) [inline CHANGELOG.md]: Please remove the mention of Daedalus process names since that is remove now

Comment (nikolaglumac) [reply CHANGELOG.md]: Good catch @DominikGuzei!
I've fixed it now ;)

**Pull request merged**